src/index.ts

```diff
@@ -0,0 +1,415 @@
+#!/usr/bin/env node
+
+import { Server } from "@modelcontextprotocol/sdk/server/index.js";
```

Comment:
CAn we break this file into multiple smaller files bases on their utility.
Like modules, utils etc.